### PR TITLE
Fixes #108

### LINKED ISSUE #108: Binary decode bug in decoding registered concrete type into concrete var.
PR incoming.

### PR DESCRIPTION
**Pull request by jaekwon**

### CODE REVIEW

**Review by odeke-em — APPROVED** [review]
How did you catch this issue?
Thank you and LGTM!

File: binary-decode.go [hunk 1]
```diff
@@ -50,7 +50,7 @@ func (cdc *Codec) decodeReflectBinary(bz []byte, info *TypeInfo, rv reflect.Valu
 		err = errors.New("EOF skipping prefix bytes.")
 		return
 	}
-	if !info.Prefix.EqualBytes(bz) {
+	if !info.Prefix.EqualBytes(bz[:PrefixBytesLen]) {
```
Comment (odeke-em) [inline binary-decode.go]: I was wondering about this too, I found it in my translation to Unmarshaling from an io.Reader
screen shot 2018-02-23 at 11 23 14 pm

Comment (jaekwon) [reply binary-decode.go]: I was reading the code and it looked wrong.  heh.

Comment (odeke-em) [reply binary-decode.go]: High five!

**Pull request merged**